docs/config/01-configuration-file.md

```diff
@@ -67,6 +67,69 @@ require('ts-node').register({
 require('./karma.conf.ts');
 ```
 
+### Asynchronous Configuration
+
+The function exported by the configuration file may return a promise whose
+resolution will tell Karma when all configuration is complete, including the
+final call to `config.set()`.
+
+Just as with a synchronous configuration's return value not being used, an
+asynchronous configuration's resolved value will not be used. With both
+configuration types, `config.set()` MUST BE called to assign the configuration
+object.
+
+When using the command line interface, returning a promise is all that is
+needed.
+
```

Comment:
Maybe add an example to this section? Something like below to demonstrate the use case.

```javascript
// karma.conf.js
module.exports = async (config) => {
  const freePort = await findFreePort();
  config.set({
    port: freePort,
    basePath: '../..',
    frameworks: ['jasmine'],
    //...
  });
}
```